In [38]:
print("helel")

helel


In [3]:
import numpy as np 
import matplotlib.pyplot as plt 
import seaborn as sns
import pandas as pd 
%matplotlib inline 
import warnings
warnings.filterwarnings('ignore')
from pathlib import Path

In [4]:
csv_path = Path(r"C:\Users\ranji\OneDrive\Desktop\1st_mlproject\src\components\notebook\data\stud.csv")
df = pd.read_csv(csv_path)

In [41]:
df.head()

gender race_ethnicity parental_level_of_education         lunch  \
0  female        group B           bachelor's degree      standard   
1  female        group C                some college      standard   
2  female        group B             master's degree      standard   
3    male        group A          associate's degree  free/reduced   
4    male        group C                some college      standard   

  test_preparation_course  math_score  reading_score  writing_score  
0                    none          72             72             74  
1               completed          69             90             88  
2                    none          90             95             93  
3                    none          47             57             44  
4                    none          76             78             75

In [42]:
df.isna().sum()

gender                         0
race_ethnicity                 0
parental_level_of_education    0
lunch                          0
test_preparation_course        0
math_score                     0
reading_score                  0
writing_score                  0
dtype: int64

In [43]:
df.duplicated().sum()

np.int64(0)

In [44]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 8 columns):
 #   Column                       Non-Null Count  Dtype 
---  ------                       --------------  ----- 
 0   gender                       1000 non-null   object
 1   race_ethnicity               1000 non-null   object
 2   parental_level_of_education  1000 non-null   object
 3   lunch                        1000 non-null   object
 4   test_preparation_course      1000 non-null   object
 5   math_score                   1000 non-null   int64 
 6   reading_score                1000 non-null   int64 
 7   writing_score                1000 non-null   int64 
dtypes: int64(3), object(5)
memory usage: 62.6+ KB


In [45]:
df.nunique()

gender                          2
race_ethnicity                  5
parental_level_of_education     6
lunch                           2
test_preparation_course         2
math_score                     81
reading_score                  72
writing_score                  77
dtype: int64

In [46]:
df.describe()

math_score  reading_score  writing_score
count  1000.00000    1000.000000    1000.000000
mean     66.08900      69.169000      68.054000
std      15.16308      14.600192      15.195657
min       0.00000      17.000000      10.000000
25%      57.00000      59.000000      57.750000
50%      66.00000      70.000000      69.000000
75%      77.00000      79.000000      79.000000
max     100.00000     100.000000     100.000000

#insight 

From above description of numerical data, all means are very close to each other - between 66 and 68.05;
all the standard deviation are closed -between 14.6 and 15.19;

In [1]:
print("Categories in 'gender' variable: ", end = " ")
print(df['gender'].unique())

print("Categories in 'race_ethnicity' variable: ", end = "")
print(df['race_ethnicity'].unique())

print("Categories in 'parental_level_of_education' variable: ", end = "")
print(df['parental_level_of_education'].unique())

print("Categories in 'lunch' variable: ", end = "")
print(df['lunch'].unique())

print("Categories in 'test_preparation_course' variable: ", end = "")
print(df['test_preparation_course'].unique())

Categories in 'gender' variable:  

NameError: name 'df' is not defined

In [48]:
numeric_features = [feature for feature in df.columns if df[feature].dtype != 'O']
categorical_features = [feature for feature in df.columns if df[feature].dtype == 'O']

print('we have {} numerical features : {}'.format(len(numeric_features), numeric_features))
print('we have {} categorical features: {}'.format(len(categorical_features), categorical_features))

we have 3 numerical features : ['math_score', 'reading_score', 'writing_score']
we have 5 categorical features: ['gender', 'race_ethnicity', 'parental_level_of_education', 'lunch', 'test_preparation_course']


#Adding columns for "Total Score" and "Average"

In [49]:
df['total_score'] = df['math_score'] + df['reading_score'] + df['writing_score']
df['average_score'] = df['total_score'] / 3
df.head()

gender race_ethnicity parental_level_of_education         lunch  \
0  female        group B           bachelor's degree      standard   
1  female        group C                some college      standard   
2  female        group B             master's degree      standard   
3    male        group A          associate's degree  free/reduced   
4    male        group C                some college      standard   

  test_preparation_course  math_score  reading_score  writing_score  \
0                    none          72             72             74   
1               completed          69             90             88   
2                    none          90             95             93   
3                    none          47             57             44   
4                    none          76             78             75   

   total_score  average_score  
0          218      72.666667  
1          247      82.333333  
2          278      92.666667  
3          148      49.333333  
4          229      76.333333

In [50]:
reading_full = df[df['reading_score'] == 100]['average_score'].count()
writing_full = df[df['writing_score'] == 100]['average_score'].count()
math_full = df[df['math_score'] == 100]['average_score'].count()
print(f'No of students with full marks in reading: {reading_full}')
print(f'No of students with full marks in writing: {writing_full}')
print(f'No of students with full marks in math: {math_full}')

No of students with full marks in reading: 17
No of students with full marks in writing: 14
No of students with full marks in math: 7


In [51]:
reading_less_20 = df[df['reading_score'] <=20]['average_score'].count()
writing_less_20 = df[df['writing_score'] <=20]['average_score'].count()
math_less_20 = df[df['math_score'] <=20]['average_score'].count()
print(f'no of students with less than or equal to 20 marks in reading: {reading_less_20}')
print(f'no of students with less than or equal to 20 marks in writing: {writing_less_20}')
print(f'no of students with less than or equal to 20 marks in math: {math_less_20}')

In [2]:
fig, axs = plt.subplots(1,2, figsize=(15,7))
plt.subplot(121)
sns.histplot(df['average_score'], bins=30, kde=True, color='g')
plt.subplot(122)
sns.histplot(data=df,x = 'average_score', kde = True, hue = 'gender')
plt.show()

NameError: name 'plt' is not defined

In [53]:
fig, axs = plt.subplots(1,2, figsize=(15,7))
plt.subplot(121)
sns.histplot(df['total_score'], bins=30, kde=True, color='g')
plt.subplot(122)
sns.histplot(data=df,x = 'total_score', kde = True, hue = 'gender')
plt.show()

In [54]:
plt.subplots(1,3, figsize = (25,6))
plt.subplot(141)
sns.histplot(data = df, x = 'average_score', kde = True, hue = 'lunch')
plt.subplot(142)
sns.histplot(data = df[df.gender =='female'], x = 'average_score', kde = True, hue ='lunch')
plt.subplot(143)
sns.histplot(data = df[df.gender =='male'], x = 'average_score', kde = True, hue ='lunch')
plt.show()

In [ ]:
plt.subplots(1,3,figsize=(25,6))
plt.subplot(141)
ax =sns.histplot(data=df,x='average_score',kde=True,hue='parental_level_of_education')
plt.subplot(142)
ax =sns.histplot(data=df[df.gender=='male'],x='average_score',kde=True,hue='parental_level_of_education')
plt.subplot(143)
ax =sns.histplot(data=df[df.gender=='female'],x='average_score',kde=True,hue='parental_level_of_education')
plt.show()

In [54]:
plt.subplots(1,3,figsize=(25,6))
plt.subplot(141)
ax =sns.histplot(data=df,x='average_score',kde=True,hue='race_ethnicity')
plt.subplot(142)
ax =sns.histplot(data=df[df.gender=='female'],x='average_score',kde=True,hue='race_ethnicity')
plt.subplot(143)
ax =sns.histplot(data=df[df.gender=='male'],x='average_score',kde=True,hue='race_ethnicity')
plt.show()